docs/DeveloperGuide.md

```diff
@@ -1,38 +1,150 @@
 # Developer Guide
 
 ## Acknowledgements
+Ideas for this project partly comes from NUSmods at https://nusmods.com/
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+API and library used includes the following:
+- [NUSMods API](https://api.nusmods.com/v2/) - To extract information about modules in NUS
+- [FasterXML](https://github.com/FasterXML/jackson) - Helps parse json files from the NUSMods API
 
 ## Design & implementation
+#### Architecture
+<img src="images/architectureDiagram.png" width="280" />
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+
+The ***Architecture Diagram*** given above presents the high-level design of the product. The main components included are as follows:
+1. `Duke`: Initialised when launch and acts as the main logic component of the program.
+2. `UI`: Handles all user interaction, including printing responses and receiving user inputs.
+3. `Parser`: Parses main user commands and calls for appropriate functions.
+4. `commands`: Exists as a package which includes all command functions.
```

Comment:
Do consider explaining what command does instead of mentioning what it packages.